clang: Missing function prototype warnings for zlib since jpeg/png->stb replacement

clang issues many deprecation warnings for zlib since commit 2262599707f68e2f480f28064928c2ecb0eea1e6.

For example:

```
/Users/steve/XcodeProjects/RBDOOM-3-BFG/neo/libs/zlib/adler32.c:65:15: warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
uLong ZEXPORT adler32(adler, buf, len)
              ^

and so on for many zlib source files...
```

This warning was previously suppressed for minizip, but re-emerged for zlib itself following the restructuring.

I will submit a PR to suppress this one for now.  Perhaps zlib/minizip should be updated in the future.

--- a/neo/CMakeLists.txt
+++ b/neo/CMakeLists.txt
@@ -1534,6 +1534,7 @@ else()
 
 	# SRS - disable certain gcc/clang warnings for select third-party source libraries, consider updating versions in the future?
 	set_source_files_properties(
+			${ZLIB_SOURCES}
 			${MINIZIP_SOURCES}
 			PROPERTIES
 			COMPILE_FLAGS "-Wno-stringop-overread -Wno-deprecated-non-prototype"

